biogrid/libs/biogrid-simulator/src/lib/biobrain/biobrain-class.ts

```diff
@@ -243,16 +318,117 @@ export class BioBrain implements Brain {
+    };
+  }
+
+  /**
+   * This function is used to determine the resistance of the wires used for transporting power from one end to another
+   * @param length is the distance travelled by the power from supplier to receiver
+   */
+  private determineResistanceInWires(length: number) {
+    return calculateResistance(length);
+  }
+
+  /**
+   * This method determmines which grid item is in the shortest path.
+   * @param start is the starting gridItem name in the shortestPath
+   * @param dest is gridItem name for the destination in the shortestPath
+   * @param shortestDistances holds an object of key, value pair of vertex -> adj vertices with their shortest distance to the key vertex
+   * @returns a list of the names for the gridItems in the shortestPath
+   */
+  private determineGridItemsInShortestPath(
+    start: string,
+    dest: string,
+    shortestDistances: ShortestDistances
+  ): string[] {
+    // Keeps track of the gridItems through which the power is going to pass
+    let gridItemsInShortestPath = [dest];
+    // Keeps track of the parent / predecessor of a vertex in the shortest distance path
+    let parent = shortestDistances[start][dest].predecessor;
+    // Check to see if parent is not start
+    while (parent !== start) {
+      gridItemsInShortestPath.push(parent);
+      parent = shortestDistances[start][parent].predecessor;
+    }
+    gridItemsInShortestPath.push(start);
+    return gridItemsInShortestPath;
+  }
+
+/**
+ * This method determmines the resistance of the gridItems in the shortest path.
+ * @param start is the starting gridItem name in the shortestPath
+ * @param dest is gridItem name for the destination in the shortestPath
+ * @param shortestDistances holds an object of key, value pair of vertex -> adj vertices with their shortest distance to the key vertex
+ * @returns the total resistance of the gridItems in the shortest path excluding the wires (shortest distance itself)
+ */
+  private determineResistanceInShortestPath(
+    start: string,
+    dest: string,
+    shortestDistances: ShortestDistances
+  ) {
+    const gridItemsInShortestPath 
+      = this.determineGridItemsInShortestPath(start, dest, shortestDistances);
+    let gridItemResistance = 0;
+    for (const gridItemName of gridItemsInShortestPath) {
+      const gridItem: GridItem = this.clonedGraph.node(gridItemName);
+      gridItemResistance += gridItem.gridItemResistance;
     }
-    return supplyToSupplyFromAgents;
+    return gridItemResistance;
+  }
+
+  /**
+   * This method calculates the voltage to be received
+   * @param power is the power which is to be received by the receiving gridItem
+   * @param resistance is the the resistance of the receiving gridItem
+   * @returns the voltage for a particular griditem whose @param power and @param resistance are given
+   */
+  private calculateVoltageToBeReceived(power: Power, resistance: Resistance) {
+    return calculateVoltageFromPower(power, resistance);
```

Comment:
why have two methods? can you just remove calculateVoltageToBeReceived and call calculateVoltageFromPower directly?